elements/pl-string-input/pl-string-input.py

```diff
@@ -0,0 +1,228 @@
+import lxml.html
+from html import escape
+import chevron
+import math
+import prairielearn as pl
+import random
+
+
+def prepare(element_html, element_index, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    required_attribs = ['answers_name']
+    optional_attribs = ['weight', 'correct_answer', 'label', 'suffix', 'display', 'remove_leading_trailing', 'remove_between_spaces', 'allow_blank']
+    pl.check_attribs(element, required_attribs, optional_attribs)
+
+    name = pl.get_string_attrib(element, 'answers_name')
+    correct_answer = pl.get_string_attrib(element, 'correct_answer', None)
+
+    if correct_answer is not None:
+        if name in data['correct_answers']:
+            raise Exception('duplicate correct_answers variable name: %s' % name)
+        data['correct_answers'][name] = correct_answer
+
+
+def render(element_html, element_index, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, 'answers_name')
+    label = pl.get_string_attrib(element, 'label', None)
+    suffix = pl.get_string_attrib(element, 'suffix', None)
+    display = pl.get_string_attrib(element, 'display', 'inline')
+    remove_leading_trailing = pl.get_string_attrib(element, 'remove_leading_trailing', False)
+    remove_between_spaces = pl.get_string_attrib(element, 'remove_between_spaces', False)
+
+    if data['panel'] == 'question':
+        editable = data['editable']
+        raw_submitted_answer = data['raw_submitted_answers'].get(name, None)
+
+        # Get info strings
+        info_params = {'format': True}
+        with open('pl-string-input.mustache', 'r', encoding='utf-8') as f:
+            info = chevron.render(f, info_params).strip()
+        with open('pl-string-input.mustache', 'r', encoding='utf-8') as f:
```

Comment:
It would be a bit more efficient to not read the file twice. I think you can do:
```
with open(...) as f:
    template = f.read()
    info = chevron.render(template, info_params).strip()
    ...
    shortinfo = chevron.render(template, info_params).strip()
```